src/config/flat/typescript.ts

```diff
@@ -0,0 +1,35 @@
+import type { PluginFlatBaseConfig } from '../../types'
+
+/**
+ * This config:
+ * 1) adds `.jsx`, `.ts`, `.cts`, `.mts`, and `.tsx` as an extension
+ * 2) enables JSX/TSX parsing
+ */
+
+// Omit `.d.ts` because 1) TypeScript compilation already confirms that
+// types are resolved, and 2) it would mask an unresolved
+// `.ts`/`.tsx`/`.js`/`.jsx` implementation.
+const typeScriptExtensions = ['.ts', '.tsx', '.cts', '.mts'] as const
+
+const allExtensions = [...typeScriptExtensions, '.js', '.jsx', '.cjs', '.mjs'] as const
+
+export default {
+  settings: {
+    'import-x/extensions': allExtensions,
+    'import-x/external-module-folders': ['node_modules', 'node_modules/@types'],
+    'import-x/parsers': {
+      '@typescript-eslint/parser': [...typeScriptExtensions],
+    },
+    'import-x/resolver': {
+      node: {
```

Comment:
i think you also need to add "typescript: true" above this line in order for the no-cycle rule to work correctly

Reply:
(i found this from [this issue](https://github.com/un-ts/eslint-plugin-import-x/issues/29#issuecomment-2119219537))

Reply:
Would doing that not add a dependency on https://github.com/import-js/eslint-import-resolver-typescript?

FWIW, I use only `typescript: true` on my personal config and had no issues linting node scripts so far.

Reply:
Oh, I was under the impression that `[eslint-import-resolver-typescript](https://github.com/import-js/eslint-import-resolver-typescript)` was mandatory for `eslint-plugin-import-x` + TypeScript. Is that not the case after this PR? If not, ignore my original suggestion.

Reply:
I think `eslint-import-resolver-typescript` is currently only mandatory for typescript, but I think we should make it a dependency and default to it for both JS and TS. The default JS resolver has too many issues like not supporting `exports` object format which are resolved in the typescript resolver.

Reply:
huge +1 to that. abstract away the complexity for the ned-user, have this plugin "just work"

Reply:
Wouldn't this be more appropriate as a separate change?  I was aiming for parity between existing configs and the new flat configs.

Reply:
> Wouldn't this be more appropriate as a separate change? I was aiming for parity between existing configs and the new flat configs.

Let's do a separate PR!